Anarchist Library.js

```diff
@@ -0,0 +1,452 @@
+	"Plain PDF": { ext: ".pdf", mimeType: "application/pdf" },
+	"A4 PDF": { ext: ".a4.pdf", mimeType: "application/pdf" },
+	"Letter PDF": { ext: ".lt.pdf", mimeType: "application/pdf" },
+	EPub: { ext: ".eupb", mimeType: "application/epub+zip" },
+	"Printer-friendly HTML": { ext: ".html", mimeType: "text/html" },
+	LaTeX: { ext: ".tex", mimeType: "application/x-tex" },
+	"Plain Text": { ext: ".muse", mimeType: "text/plain" },
+	"Source Zip:": { ext: ".zip", mimeType: "application/zip" },
+	Snapshot: { ext: "snapshot", mimeType: "text/html" }
+};
+
+function getSearchResults(doc, checkOnly) {
+	let items = {};
+	let results = doc.querySelectorAll('a.list-group-item');
+	for (let i = 0; i < results.length; i++) {
+		if (checkOnly) {
+			return true;
+		}
+		items[results[i].href] = ZU.trimInternal(text(results[i], "strong"));
+	}
+	return items;
+}
+
+async function scrape(doc, url = doc.location.href) {
+	// ToDo: get fancier here, allow other types
+	let item = new Zotero.Item('webpage');
+
+	// These may be expanded on in the future
+	let attachmentTypes = {
+		PDF: allAttachmentTypes["Plain PDF"],
+	};
+
+	item.accessed = new Date().toString();
+	item.url = url;
+	item.language = attr(doc, "html", "lang");
+
+	let itemType = attr(doc, '[property~="og:type"]', 'content');
+	let tagNodeList = doc.querySelectorAll(`[property~="og:${itemType}:tag"]`);
+	let description = attr(doc, '[property~="og:description"]', 'content');
+	let author = attr(doc, `[property~="og:${itemType}:author"]`, 'content');
+	item.creators.push(ZU.cleanAuthor(author, "author"));
+
+	if (description) {
+		item.description = description;
+		// misses https://theanarchistlibrary.org/library/leo-tolstoy-the-complete-works-of-count-tolstoy-volume-12
+		let re = /(?<=[Tt]ranslated(?: +to [Ee]nglish)? +by ).*$/u;
+		let translatedMatch = description.match(re);
+		if (translatedMatch) {
+			item.creators.push(ZU.cleanAuthor(translatedMatch[0], "translator", translatedMatch[0].includes(",")));
+		}
+	}
+
+	let date = getPreambleVal(doc, "textdate");
+	let notes = getPreambleVal(doc, "preamblenotes");
+	// misses link here: https://theanarchistlibrary.org/library/margaret-killjoy-it-s-time-to-build-resilient-communities
+	let source = getPreambleVal(doc, "preamblesrc");
+
+	for (let tagNode of tagNodeList) {
+		item.tags.push({ tag: tagNode.content });
+	}
+
+	let title = attr(doc.head, '[property~="og:title"][content]', 'content');
+	item.title = title;
+	item.date = date;
+	if (notes) {
+		item.notes.push({ note: ZU.trimInternal(notes) });
+	}
+	if (source) {
+		item.notes.push({ note: `Source: ${ZU.trimInternal(source)}` });
+	}
+
+	for (const [typeName, typeInfo] of Object.entries(attachmentTypes)) {
+		let attachment = {
+			title: typeName,
+			url: `${doc.location.href}${typeInfo.ext}`,
+			mimeType: typeInfo.mimeType
+		};
+
+		if (typeInfo.ext == "snapshot") {
+			attachment.snapshot = true;
+		}
+
+		item.attachments.push(attachment);
+	}
+
+	return item.complete();
```

Comment:
`complete()` only returns a promise in async import translation (it's complicated), and `scrape` doesn't need to return anything, so:

```suggestion
	item.complete();
```